edgecenter/resource_edgecenter_volume.go

```diff
@@ -247,51 +236,50 @@ func resourceVolumeUpdate(ctx context.Context, d *schema.ResourceData, m interfa
 	volumeID := d.Id()
 	log.Printf("[DEBUG] Volume id = %s", volumeID)
 	config := m.(*Config)
-	provider := config.Provider
-	client, err := CreateClient(provider, d, VolumesPoint, VersionPointV1)
+	clientV2 := config.CloudClient
+
+	regionID, projectID, err := GetRegionIDandProjectID(ctx, clientV2, d)
 	if err != nil {
 		return diag.FromErr(err)
 	}
 
-	volume, err := volumes.Get(client, volumeID).Extract()
+	clientV2.Region = regionID
+	clientV2.Project = projectID
+
+	volume, _, err := clientV2.Volumes.Get(ctx, volumeID)
 	if err != nil {
 		return diag.FromErr(err)
 	}
 
 	if d.HasChange("name") {
-		name := d.Get("name").(string)
-		_, err := volumes.Update(client, volumeID, volumes.UpdateOpts{Name: name}).Extract()
+		newName := d.Get("name").(string)
+		_, _, err := clientV2.Volumes.Rename(ctx, volumeID, &edgecloudV2.Name{Name: newName})
 		if err != nil {
 			return diag.FromErr(err)
 		}
 	}
 
 	if d.HasChange("size") {
-		newValue := d.Get("size")
-		newSize := newValue.(int)
+		newSize := d.Get("size").(int)
 		if newSize != 0 {
 			if volume.Size < newSize {
-				err = ExtendVolume(client, volumeID, newSize)
+				task, _, err := clientV2.Volumes.Extend(ctx, volumeID, &edgecloudV2.VolumeExtendSizeRequest{Size: newSize})
 				if err != nil {
 					return diag.FromErr(err)
 				}
-			} else {
-				return diag.Errorf("Validation error: unable to update size field because new volume size must be greater than current size")
+				if err = utilV2.WaitForTaskComplete(ctx, clientV2, task.Tasks[0]); err != nil {
+					return diag.FromErr(err)
+				}
```

Comment:
добавил в конце if
`else {
				return diag.Errorf("Validation error: unable to update size field because new volume size must be greater than current size")
}`